### DISCUSSION TITLE
Change object x to y

### DISCUSSION THREAD

**Discussion by TheLarry98** [OpenRCT2/OpenRCT2#15671] [Feature requests] [answered]
I currently work on some RCT1 modified parks with for example winter instead of a summer map or a desert instead of a winter map, so it is really hard to replace always the same thing over and over.

So it would be good when there would be a option to replace one typ of object to another. (This should affect the complete map)

Here are some examples.
1. Change all Red Fire Tree to the Snow Red Fir Trees
2. Change the Tarmac Footpath to the Ash Footpath

I hope you guys understand what I mean.

**Comment by Gymnasiast** [comment]
This would be a very useful feature for cleaning up parks, especially when coupled with a way to unload the old object if desired. Then it can be used to replace objects with a fixed version, or replacing the entrance without removing all the existing ones in advance.

**Comment by PoshTopHat** [comment]
If you could combine it with the option to exchange something for nothing (i.e. getting rid of only one type of tree) for the whole map, I think RCT map creators would build a statue in your honour, too.

**Comment by AaronVanGeffen** [comment]
Maybe this would work best as a console command?

**Comment by Gymnasiast** [comment]
@AaronVanGeffen : The first version will probably be a console command, though I would want a GUI for it as well eventually.

**Comment by Liampie** [comment]
Note: it's important that you can change object x to y from the outside, so without opening the park, for reasons specified here:

https://github.com/OpenRCT2/OpenRCT2/issues/5821

**Comment by Richard-L** [comment]
After [yesterday's exchange in the #plugin channel on discord](https://discord.com/channels/264137540670324737/696130778618396683/720749243396259850), I've worked in some of the feedback into what I'd suggest as a "Scenery Replacer" menu or plugin for this case:
![scenery-replacer](https://user-images.githubusercontent.com/4934209/84548338-ab622880-ad05-11ea-9aaa-7edd3d9ba6f7.png)

**Top**
User is chosing the object they want to replace.

**Bottom**
A pair for replacement is chosen. Log output ist just a mock-up.

The livesearch text input would try to match data in "Identifier" and "Menu Name".

- Slot #: the 720 slots classic .sv6 offers, or later more with .park. Can sort asc/desc. Here it's set to "ascending".
- Identifier: self-explanatory. Can sort asc/desc alphabetically.
- Count: the number of instances of the object on the map. Can sort asc/desc. People tell me they often try to find objects they've only used very few times so they know what to delete in order to free up a slot.
- Type: drop-down filter for: Attraction, Small/Large Scenery, Wall, … and an "All Types" aggregate.
- Source: drop-down filter for: RCT2, WW, TT, ORCT2, Custom, No WWTT, and "All Sources".
- Tile: drop-down filter for: 1/4, 1x1, 2x2, ... 7x7. Once a choice is made on either side, the other side greys out livesearch results of different tile size.
- Menu Name: the object name users might be more familiar with and thus use to search. Can sort alphabetically.

Happy to work in more feedback. Would be so exciting to see this realised!

**Comment by Sadret** [comment, marked as answer]
A lot of the functionality that was discussed in this thread is now possible with the Scenery Manager plugin:
http://openrct2plugins.org/plugin/MDEwOlJlcG9zaXRvcnkzMTQyNzU2NDQ=/openrct2-scenery-manager
https://github.com/Sadret/openrct2-scenery-manager
